### PR TITLE
[MINOR] EDA-1264 Create tournament view for new tournament

### PR DESCRIPTION
**Pull request by vgatica-eb**
A CreatedTournamentView was created. Also a template with a js file.

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -53,15 +54,16 @@
             <li class="nav-item">
                 <a class="nav-link collapsed" href="#" data-toggle="collapse" data-target="#collapseTwo"
                     aria-expanded="true" aria-controls="collapseTwo">
-                    <i class="fas fa-fw fa-cog"></i>
-                    <span>Development</span>
+                    <!-- <i class="fas fa-fw fa-cog"></i> -->
```
Comment (ibrizuela-eb) [inline templates/base.html:57]: Remove comments if you don't need them

File: templates/base.html [hunk 2]
```diff
@@ -53,15 +54,16 @@
...
+                    <span>MENU</span>
                 </a>
                 <div id="collapseTwo" class="collapse" aria-labelledby="headingTwo" data-parent="#accordionSidebar">
                     <div class="bg-white py-2 collapse-inner rounded">
                         <!-- <h6 class="collapse-header">Development:</h6> -->
```
Comment (ibrizuela-eb) [inline templates/base.html:62]: Remove comments if you don't need them

File: tournaments/forms.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class TournamentForm(forms.Form):
+    tournament = forms.CharField(label='name')
+    bots_selected = forms.CharField(label='botsSelected', widget=forms.HiddenInput(), required=False)
+    bots = forms.ChoiceField(label='Bots', widget=forms.HiddenInput(), choices=[], required=False)
+
+    def setup_bots_choices(self):
+        bots = Bot.objects.all()
+        self.fields['bots'].choices = enumerate([bot.name for bot in bots])
```
Comment (ibrizuela-eb) [inline tournaments/forms.py:12]: Why you select all Bots in DB? I think that you must use only online bots and you've already had 2 methods to do that in `bot_handler.py`.

1. get_users_data() # returns a list with online bots
2. get_online_bots(users) # returns a enumerate list of users: List.

File: tournaments/views.py [hunk 4]
```diff
@@ -1,3 +1,58 @@
...
+                '{} successfully added'.format(data[0])
+            )
+        else:
+            messages.add_message(
+                self.request,
+                messages.ERROR,
+                'It is not possible to create this record, a tournament already exists with the name '
+                '{}. Try a new name'.format(data[0])
+            )
+            self.success_url = reverse_lazy('tournaments:create_tournament')
```
Comment (ibrizuela-eb) [inline tournaments/views.py:50]: You've already defined the success_url in line `21`.

File: templates/base.html (hunk 1 above)
Comment (vgatica-eb) [reply templates/base.html:57]: done

File: templates/base.html (hunk 2 above)
Comment (vgatica-eb) [reply templates/base.html:62]: done

File: tournaments/views.py (hunk 4 above)
Comment (vgatica-eb) [reply tournaments/views.py:50]: done

File: tournaments/forms.py (hunk 3 above)
Comment (vgatica-eb) [reply tournaments/forms.py:12]: As we were talking, we should have a meet with the PM to define the values he want to show. I set by default all bot of database, anyway we can change it after... with a refactor

File: edagames/urls.py [hunk 5]
```diff
@@ -23,4 +23,5 @@
     path('oauth/', include('social_django.urls', namespace='social')),
     path('', include(('development.urls', 'auth'), namespace='development')),
     path('', include('development.urls', namespace='api')),
+    path('', include(('tournaments.urls'), namespace='tournaments')),
```
Comment (arielaranda-eb) [inline edagames/urls.py:26]: nit: unnecessary parethesis?

File: tournaments/views.py [hunk 6]
```diff
@@ -1,3 +1,58 @@
...
+from django.urls import reverse_lazy
+from django.contrib import messages
+from django.contrib.auth.mixins import (
+    LoginRequiredMixin,
+    UserPassesTestMixin,
+)
 
-# Create your views here.
+
+class StaffRequiredMixin(LoginRequiredMixin, UserPassesTestMixin):
```
Comment (arielaranda-eb) [inline tournaments/views.py:12]: maybe this mixin could go in the mixin module

File: edagames/urls.py (hunk 5 above)
Comment (vgatica-eb) [reply edagames/urls.py:26]: removed

File: tournaments/views.py [hunk 7]
```diff
@@ -1,3 +1,58 @@
...
+        if self.request.method == 'POST':
+            form = TournamentForm(data=self.request.POST)
+        else:
+            form = TournamentForm()
+        form.setup_bots_choices()
+        return form
+
+    def form_valid(self, form):
+        data = []
+        if form.is_valid():
```
[the remark this answers is not in the document]
Comment (vgatica-eb) [reply tournaments/views.py:33]: yes, that's true. Now it's removed. Thanks

**Review by ibrizuela-eb — APPROVED** [review]

**Pull request merged**